src/fuse_reduce.cpp

```diff
@@ -188,8 +189,9 @@ static void create_reduce_modules(module_pass_manager& mpm)
 template <class... Ms>
 static auto match_broadcast(Ms... ms)
 {
+    // todo:  does this change here make sense?
     return match::skip(match::name("contiguous"))(
-               match::name("multibroadcast")(match::arg(0)(ms...), match::used_once())
+               match::name("multibroadcast")(match::arg(0)(ms...), match::used_once(), match::broadcastable())
```

Comment:
It shouldn't be checking if its broadcastable(of course its broadcastable since its broadcasting it in the graph). It should just check that the input ndim matches the output ndim.

Reply:
done.